sql/threadpool_unix.cc

```diff
@@ -1344,6 +1344,7 @@ bool Thread_pool_connection_handler::add_connection(Channel_info *channel_info)
 
   if (unlikely(!connection))
   {
+    my_free(thd->get_protocol_classic()->get_vio());
```

Comment:
@altmannmarcelo From what I see from other parts of the code, for instance here (https://github.com/percona/percona-server/blob/c08e5cd44eac25634931eea27a2081d41f2398c3/plugin/innodb_memcached/innodb_memcache/src/handler_api.cc#L380)
In patterns like this, we usually call `thd->release_resources();` instead.
If this does not work, memory from `vio` object should not be deleted directly by `my_free()`. Use `vio_delete()` instead